e2e_tests/tests/cluster/test_checkpoints.py

```diff
@@ -45,6 +48,75 @@ def test_gc_checkpoints_lfs() -> None:
     run_gc_checkpoints_test(exp.shared_fs_checkpoint_config())
 
 
+@pytest.mark.e2e_cpu
+def test_delete_checkpoints() -> None:
+    base_conf_path = conf.fixtures_path("no_op/single-default-ckpt.yaml")
+    config = conf.load_config(str(base_conf_path))
+    config["checkpoint_storage"] = {
+        "type": "shared_fs",
+        "host_path": "/tmp",
+        "storage_path": "delete-checkpoints-e2etest",
+    }
+    config["min_checkpoint_period"] = {"batches": 10}
+
+    exp_id_1 = exp.run_basic_test_with_temp_config(
+        config, model_def_path=conf.fixtures_path("no_op"), expected_trials=1
+    )
+
+    exp_id_2 = exp.run_basic_test_with_temp_config(
+        config, model_def_path=conf.fixtures_path("no_op"), expected_trials=1
+    )
+
+    test_session = exp.test_session()
+    exp_1_checkpoints = bindings.get_GetExperimentCheckpoints(
+        session=test_session, id=exp_id_1
+    ).checkpoints
+    exp_2_checkpoints = bindings.get_GetExperimentCheckpoints(
+        session=test_session, id=exp_id_1
+    ).checkpoints
+    assert len(exp_1_checkpoints) > 0, f"no checkpoints found in experiment with ID:{exp_id_1}"
+    assert len(exp_2_checkpoints) > 0, f"no checkpoints found in experiment with ID:{exp_id_2}"
+
+    indices_exp_1 = [i for i in range(0, len(exp_1_checkpoints))]
+    indices_exp_2 = [i for i in range(0, len(exp_2_checkpoints))]
+    d_exp_1_checkpoint_uuids = [
+        exp_1_checkpoints[d_index].uuid for d_index in random.sample(indices_exp_1, 2)
+    ]
+    d_exp_2_checkpoint_uuids = [
+        exp_2_checkpoints[d_index].uuid for d_index in random.sample(indices_exp_2, 2)
+    ]
+
+    d_checkpoint_uuids = d_exp_1_checkpoint_uuids + d_exp_2_checkpoint_uuids
+    delete_body = bindings.v1DeleteCheckpointsRequest(checkpointUuids=d_checkpoint_uuids)
+    bindings.delete_DeleteCheckpoints(session=test_session, body=delete_body)
+    wait_for_gc_to_finish(exp_id_1)
+    wait_for_gc_to_finish(exp_id_2)
+
+    for d_c in d_checkpoint_uuids:
+        ensure_checkpoint_deleted(test_session, config, d_c)
+
+
+def ensure_checkpoint_deleted(test_session, config, d_checkpoint_uuid) -> bool:
+    d_checkpoint = bindings.get_GetCheckpoint(
+        session=test_session, checkpointUuid=d_checkpoint_uuid
+    ).checkpoint
+    if d_checkpoint is not None:
+        assert (
+            d_checkpoint.state == determinedcheckpointv1State.STATE_DELETED
+        ), f"checkpoint with uuid {d_checkpoint_uuid} does not have a deleted state"
+    else:
+        raise Exception(
```

Comment:
here and elsewhere, maybe a good thing to use `pytest.fail` instead of `raise` to distinguish test failures from test exceptions. though we don't really hold this up all over, i was just thinking about it in tests i was writing recently.